buildSrc/src/main/java/org/elasticsearch/gradle/testclusters/TestClusterConfiguration.java

```diff
@@ -131,7 +131,17 @@ default void waitForConditions(
                 if (lastException == null) {
                     throw new TestClustersException(message);
                 } else {
-                    throw new TestClustersException(message + message, lastException);
+                    String extraCause = "";
+                    Throwable cause = lastException;
+                    int ident = 2;
+                    while (cause != null) {
```

Comment:
I'm not sure why extracting the messages from the cause chain is necessary. If we rethrow this exception, the full stacktrace with all causes will be shown in the build scan. Can you perhaps attach an example to this PR?

Reply:
It's meant for the console not the build scan. The cause will be relevant most of the time as you want to know why it timed out, so it's a convenience so we are not forced to look at the scan